tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -122,5 +126,77 @@ public void ChangeStreamExample3()
                 next.FullDocument.Should().Be(documents[1]);
             }
         }
+
+        [Fact]
+        public void Changestream_Example_4()
+        {
+            RequireServer.Check().Supports(Feature.AggregateAddFields);
+
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase("ChangeStreamExamples");
+            database.DropCollection("inventory");
+            var inventoryCollection = database.GetCollection<BsonDocument>("inventory");
+
+            var cancelationTokenSource = new CancellationTokenSource();
+            try
+            {
+                BsonDocument document = new BsonDocument("username", "alice");
+
+                Task.Run(() =>
+                {
+                    while (!cancelationTokenSource.IsCancellationRequested)
```

Comment:
I'm not sure about which time gap will be required in case inserting only the one recodes In case if by some reasons the watch code will be executed with some delay and our task will have been finished in this moment - our test will wait for any inserting infinitely.

In my variant we have a guarantee that our watch will have insert event in any case

Reply:
I agree that the insert has to be *after* Watch has been called. Repeatedly inserting documents until cancelled is one brute force way to ensure this. This way if any inserts happen *before* Watch is called there will always be another insert afterwards. To write this test using a single insert would require some synchronization.

I'm fine leaving it as a loop...